### ISSUE TITLE
Zitadel needs a proper dockerfile based health check

### ISSUE THREAD

**Issue by ghstahl** [zitadel/zitadel#6863]
### Preflight Checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions
- [X] I have joined the [ZITADEL chat](https://zitadel.com/chat)

### Describe your problem

Zitadel and only Zitadel knows that its healthy.   

The only thing required for my service is this following line.
```
    depends_on:
      zitadel:
        condition: service_healthy
```        
This requires that the zitadel put their healthcheck inside their Dockerfile.

No google searches for trying to figure out what healthy means for zitadel.

### Describe your ideal solution

Put zitadels healthcheck inside the DOCKERFILE

### Version

all

### Environment

Self-hosted

**Comment by shaunco** [comment]
Just to add to this, trying to use `zitadel status` results in:
```json
{
  "Start":"2023-11-05T23:57:53.80151756Z",
  "End":"2023-11-05T23:57:54.22157979Z",
  "ExitCode":0,
  "Output":"time=\"2023-11-05T23:57:54Z\" level=warning msg=\"ready check failed\" caller=\"/home/runner/work/zitadel/zitadel/cmd/ready/ready.go:35\" status=200\n"}
```

which is quite confusing - why is the exit code 0 (success) if `ready check failed`?

**Comment by shaunco** [comment]
Also, the `/debug/healthz` url is helpful in Kubernetes, but not useful in Docker, as docker can only execute stuff inside the Zitadel container and the Zitadel container is built from `scratch`, which means it is lacking `wget`, `curl`, and `/bin/sh` - all good things ... just makes `zitadel status` the only option.

**Comment by shaunco** [comment]
Looks like the log was always being written and was fixed 2 weeks ago in https://github.com/zitadel/zitadel/commit/b099a26a1600be6661d9358c2ffbea034e47b51c#diff-9b29d9beac3d48fbebcc2d1d6b25a481a381275356e037485cadbe63e471d1ca 

That said, I'm not clear on the difference between `/debug/ready` and `/debug/healthz` so, maybe we just need an option like `zitadel ready --and-healthy` that checks both?

**Comment by mffap** [comment]
I agree for docker, the behavior is not very clear. I also fell into the same issue. 

There is an example in the e2e folder: 
https://github.com/zitadel/zitadel/blob/bd5506494a87a2e2144d60bf04e856fe276d8f2b/e2e/config/localhost/docker-compose.yaml#L25

Basically the healthcheck needs to call the CLI instead of the endpoint.

The issue comes down to what @shaunco mentioned here: https://github.com/zitadel/zitadel/issues/6863#issuecomment-1793893548

In Docker you can't call the enpoints without curl, that's why you need to use the CLI instead or use a different image. I think in the first place the documentation must be better on this difference to avoid this issue. Then maybe there's also a technical solution.

**Comment by aDogCalledSpot** [comment]
When using the docker compose approach, most things are overwritten with environment variables making the `--config /zitadel.yaml` inconvenient. There really should be a curl or wget installed to make use of the healthz endpoint.
